matrix-api-r0.3.0.el

```diff
@@ -1303,6 +1305,27 @@ TYPING-P should be t or nil."
                (matrix-log (a-list 'fn 'matrix-typing-error-callback
                                    'args args))))))
 
+(cl-defun matrix-mark-fully-read (room)
+  "updates the fully_read marker and read receipt location for ROOM"
+  ;; spec link
+  (with-slots* (((id session timeline last-seen-event) room)
+                ((txn-id) session))
+
+    (let* ((type "m.room.message")
+           (event (car timeline))
+           (event-id (alist-get 'event_id event))
+           (data (a-list "m.fully_read" event-id
+                         "m.read" event-id))
+           (txn-id (cl-incf txn-id))
+           (room-id (url-hexify-string id))
+           (endpoint (format$ "rooms/$room-id/read_markers")))
+      (unless (eq event last-seen-event)
+        (matrix-request-request session endpoint
```

Comment:
it makes emacs unresponsive until it returns (and sometimes it doesn't return). this problem i know its because of matrix-post because when i used  matrix-request-request it worked perfectly fine (and i called it explicitly to ensure that it is causing this problem), but i reverted back to matrix-post because its not something specific to this implementation if the problem is in my system then matrix-post is ok and if not matrix-post itself should be fixed.

Reply:
the last commit fixed the issue in matrix-post, it was something related to the callback being nil. but for some reason matrix-request-request handled this case better. i didn't investigate deeper but it works fast now.